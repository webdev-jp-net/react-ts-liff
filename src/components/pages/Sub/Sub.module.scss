@use 'style/_variable' as *;

.sub {
  position: relative;
}

.menu {
  display: grid;
  gap: rem(18) rem(12);
  padding: rem(24);
}

.field {
  display: grid;
  gap: rem(18) rem(12);
  padding: rem(24);
}

.preview {
  padding: 0.5em;
  background: $c_bg;
}
