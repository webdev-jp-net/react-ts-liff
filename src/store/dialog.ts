import { createSlice } from '@reduxjs/toolkit';

type State = {
  isOpen: boolean;
  content: HTMLElement | null;
  class?: string[];
};

const initialState: State = {
  isOpen: false,
  content: null,
};

const dialog = createSlice({
  name: 'dialog',

  initialState,

  reducers: {
    updateOpen: (state, action) => {
      const result: {
        isOpen: boolean;
        class?: string[];
      } = {
        isOpen: action.payload,
      };
      if (action.payload === true) {
        // open
        document.body.classList.add('isFixedScroll');
      } else {
        // close
        document.body.classList.remove('isFixedScroll');
        result.class = [];
      }
      return {
        ...state,
        ...result,
      };
    },
    updateContent: (state, action) => {
      const result: {
        content: HTMLDivElement;
      } = {
        content: action.payload,
      };
      return {
        ...state,
        ...result,
      };
    },
    updateClass: (state, action) => {
      const result: {
        class: string[];
      } = {
        class: [state.class, ...action.payload],
      };
      return {
        ...state,
        ...result,
      };
    },
  },
});

// Action Creators
export const { updateOpen, updateContent, updateClass } = dialog.actions;

// Reducer
export default dialog.reducer;
