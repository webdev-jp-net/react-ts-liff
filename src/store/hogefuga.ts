import { createSlice } from '@reduxjs/toolkit';

type State = {
  dummyText: string;
};

const initialState: State = {
  dummyText: 'ダミーテキスト',
};

const hogefuga = createSlice({
  name: 'hogefuga',

  initialState,

  reducers: {
    updateDummyText: (state, action) => {
      const result: {
        dummyText: string;
      } = {
        dummyText: action.payload,
      };
      return {
        ...state,
        ...result,
      };
    },
  },
});

// Action Creators
export const { updateDummyText } = hogefuga.actions;

// Reducer
export default hogefuga.reducer;
