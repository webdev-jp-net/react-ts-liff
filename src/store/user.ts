import { createSlice } from '@reduxjs/toolkit';

type State = {
  userId?: string;
};

const initialState: State = {};

const user = createSlice({
  name: 'user',

  initialState,

  reducers: {
    updateUserId: (state, action) => {
      state.userId = action.payload;
    },
  },
});

// Action Creators
export const { updateUserId } = user.actions;

// Reducer
export default user.reducer;
