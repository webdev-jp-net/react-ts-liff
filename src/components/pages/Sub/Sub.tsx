/* eslint @typescript-eslint/ban-ts-comment: 0 */
import { FC, useState } from 'react';

import { useNavigate, useParams } from 'react-router-dom';

import { useSelector, useDispatch } from 'react-redux';

import { Button } from 'components/parts/Button';
import { RootState } from 'store';
import { updateDummyText } from 'store/hogefuga';

import { usePageTitle } from 'hooks/usePageTitle';

import styles from './Sub.module.scss';

// import { Hoge } from 'components/hoge';

// import { Fuga } from '@types/fuga';

// function component

export const Sub: FC = () => {
  const dispatch = useDispatch();

  const navigate = useNavigate();
  const { id } = useParams();

  const { dummyText } = useSelector((state: RootState) => state.hogefuga);

  const [tmpValue, setTmpValue] = useState<string>('');

  // テキスト更新
  const updateValue = () => {
    dispatch(updateDummyText(tmpValue));
  };

  // ページタイトル
  usePageTitle(`subページ`);

  return (
    <div className={`l-page ${styles.sub}`}>
      <h1>{id} のページ</h1>
      <p>ここへページの内容を表示</p>

      <div className={styles.menu}>
        <Button
          handleClick={() => {
            navigate('/');
          }}
        >
          HOMEへ移動
        </Button>
      </div>
      <hr />
      <div className={styles.field}>
        <p>文字をinputで書き換えられます</p>
        <p className={styles.preview}>{dummyText}</p>
        <input
          placeholder="更新ボタンで入力した文字が表示されます"
          type="text"
          onInput={e => {
            setTmpValue(e.currentTarget.value);
          }}
        />
        <Button handleClick={updateValue}>更新</Button>
      </div>
    </div>
  );
};
