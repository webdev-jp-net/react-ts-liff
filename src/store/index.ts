import { combineReducers, configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';

import hogefuga from './hogefuga';
import user from './user';

const reducer = combineReducers({
  hogefuga,
  user,
});

const middleware = getDefaultMiddleware({ serializableCheck: false });
export const store = configureStore({ reducer, middleware });

export type RootState = ReturnType<typeof reducer>;
