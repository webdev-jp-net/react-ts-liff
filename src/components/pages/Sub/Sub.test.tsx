import { Router } from 'react-router-dom';

import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { store } from 'store';

import { Sub } from './Sub';

const history = createMemoryHistory();
describe('Sub', () => {
  beforeEach(() => {
    window.scrollTo = jest.fn();
  });

  it('Subのスナップショット', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router location={history.location} navigator={history}>
          <Sub />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
