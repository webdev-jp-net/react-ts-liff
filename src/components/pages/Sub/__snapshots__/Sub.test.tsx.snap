// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Subのスナップショット 1`] = `
<DocumentFragment>
  <div
    class="l-page sub"
  >
    <h1>
       のページ
    </h1>
    <p>
      ここへページの内容を表示
    </p>
    <div
      class="menu"
    >
      <button
        class="button --basic"
        type="button"
      >
        HOMEへ移動
      </button>
    </div>
    <hr />
    <div
      class="field"
    >
      <p>
        文字をinputで書き換えられます
      </p>
      <p
        class="preview"
      >
        ダミーテキスト
      </p>
      <input
        placeholder="更新ボタンで入力した文字が表示されます"
        type="text"
      />
      <button
        class="button --basic"
        type="button"
      >
        更新
      </button>
    </div>
  </div>
</DocumentFragment>
`;
